lib/topological_inventory/ansible_tower/operations/source.rb

```diff
@@ -43,14 +47,23 @@ def params_missing?
           is_missing = false
           required_params.each do |attr|
             if (is_missing = params[attr].blank?)
-              logger.error("Source#availability_check - Missing #{attr} for the availability_check request")
+              logger.error("Source#availability_check - Missing #{attr} for the availability_check request [Source ID: #{source_id}]")
               break
             end
           end
 
           is_missing
         end
 
+        def checked_recently?
+          return false if endpoint.nil?
```

Comment:
On prod currently we have sources without an `Endpoint`.  They have `Application` records.

How would we know if the `source.applications.first.last_checked_at` was `checked_recently?`

Adam's comment was saying the same I believe (https://github.com/RedHatInsights/sources-api/issues/206#issue-587773644)

Though I see in the PR description, this was only to add the additional updates to `Endpoint` so we could make the additional changes with follow up PRs

Reply:
We are not able to do availability check for sources without endpoint now. 
Supporting it should be new feature (and better to solve it as PR following  https://github.com/RedHatInsights/topological_inventory-providers-common/pull/25)

Reply:
I guess availability check of applications is not a responsibility of our operation, should be checked externally: https://github.com/RedHatInsights/sources-api/blob/ab24b6c710c8d31eadf4f5068a01d38863b8c022/app/controllers/api/v1/sources_controller.rb#L56-L55